src/python/pants/backend/python/target_types.py

```diff
@@ -1349,3 +1350,77 @@ class PythonDistribution(Target):
         See {doc_url('python-distributions')}.
         """
     )
+
+
+class SetuptoolsSCMDummySourceField(OptionalSingleSourceField):
+    """A dummy SourceField for participation in the codegen machinery."""
```

Comment:
I wasn't sure if there was a better way to do this - AFAICT the codegen machinery requires a SourceField, even though in this case there is no source. This works fine and doesn't intrude in help etc, but maybe there is a less janky way?

Reply:
Nope, this is how you do it.